src/client-side-metrics/operation-metrics-collector.ts

```diff
@@ -211,26 +211,27 @@ export class OperationMetricsCollector {
         MetricsCollectorState.OPERATION_STARTED_ATTEMPT_NOT_IN_PROGRESS;
       this.attemptCount++;
       const endTime = hrtime.bigint();
-      if (projectId && this.attemptStartTime) {
+      if (this.attemptStartTime) {
         const totalMilliseconds = Number(
           (endTime - this.attemptStartTime) / BigInt(1000000),
         );
-        OperationMetricsCollector.metricsHandlers.forEach(metricsHandler => {
-          if (metricsHandler.onAttemptComplete) {
-            metricsHandler.onAttemptComplete({
-              attemptLatency: totalMilliseconds,
-              serverLatency: this.serverTime ?? undefined,
-              connectivityErrorCount: this.connectivityErrorCount,
-              streaming: this.streamingOperation,
-              status: attemptStatus.toString(),
-              client_name: `nodejs-bigtable/${version}`,
-              metricsCollectorData: this.getMetricsCollectorData(),
-              projectId,
-            });
-          }
-        });
+        const projectId = this.tabularApiSurface.bigtable.projectId as string;
+        const options = this.tabularApiSurface.bigtable
+          .options as ClientOptions;
+        const metricsHandler = new GCPMetricsHandler(projectId, options);
```

Comment:
I really don't want us creating new handlers each time onAttemptComplete is called. 

That's a lot of extra work to do on each rpc, and we should have all the data we need to do this once per client, and share them